ethereum/core/src/main/java/tech/pegasys/teku/core/operationvalidators/OperationStateTransitionValidator.java

```diff
@@ -0,0 +1,22 @@
+/*
+ * Copyright 2020 ConsenSys AG.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package tech.pegasys.teku.core.operationvalidators;
+
+import java.util.Optional;
+import tech.pegasys.teku.datastructures.state.BeaconState;
+
+public interface OperationStateTransitionValidator<T> {
```

Comment:
This this intended to always be a single method?:
```suggestion
@FunctionalInterface
public interface OperationStateTransitionValidator<T> {
```

Reply:
Not sure. But I'll add the modifier for now. It's only for legibility purposes, right?